Mathlib/SetTheory/Ordinal/Notation/FundamentalSequence.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Data.Set.Operations
+
+/-!
+# Fundamental sequences
+
+If `x` is a successor limit in a linear order `α`, a fundamental sequence for `x` is a strictly
+monotonic sequence `ℕ → α` whose supremum is `x`.
```

Comment:
Maybe useful to add here that every element of the sequence is meant to be below `x` (otherwise I'll just take the constant sequence `x` as a fundamental sequence)

Reply:
That's implied by the phrasing "strictly monotonic".

Reply:
In that case you should give some justification for your definition on non-successor limits

Reply:
Every ordinal `o` can be said to have a fundamental sequence, if by that we mean a strictly monotonic function `Ioo (cof o) → Ioo o` with cofinal range. In fact, this is already the definition of [`Ordinal.IsFundamentalSequence`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/Cofinality.html#Ordinal.IsFundamentalSequence). This in particular implies the unique fundamental sequence for `0` is the empty function, and the unique fundamental sequence for `succ o` is `fun _ ↦ o`.

I'll add something like this as justification to the docstring.

Reply:
Yup, that's sufficient justification!